WordPress/Classes/Stores/StatsWidgetsStore.swift

```diff
@@ -33,20 +34,22 @@ class StatsWidgetsStore {
     }
 
     /// Initialize the local cache for widgets, if it does not exist
-    func initializeStatsWidgetsIfNeeded() {
-        if HomeWidgetTodayData.read() == nil {
+    @objc func initializeStatsWidgetsIfNeeded() {
+        UserDefaults(suiteName: WPAppGroupName)?.setValue(AccountHelper.defaultSiteId, forKey: AppConfiguration.Widget.Stats.userDefaultsSiteIdKey)
+
+        if HomeWidgetTodayData.read() == nil || HomeWidgetTodayData.read()?.isEmpty == true {
             DDLogInfo("StatsWidgets: Writing initialization data into HomeWidgetTodayData.plist")
             HomeWidgetTodayData.write(items: initializeHomeWidgetData(type: HomeWidgetTodayData.self))
             WidgetCenter.shared.reloadTimelines(ofKind: AppConfiguration.Widget.Stats.todayKind)
         }
 
-        if HomeWidgetThisWeekData.read() == nil {
+        if HomeWidgetThisWeekData.read() == nil || HomeWidgetThisWeekData.read()?.isEmpty == true {
             DDLogInfo("StatsWidgets: Writing initialization data into HomeWidgetThisWeekData.plist")
             HomeWidgetThisWeekData.write(items: initializeHomeWidgetData(type: HomeWidgetThisWeekData.self))
             WidgetCenter.shared.reloadTimelines(ofKind: AppConfiguration.Widget.Stats.thisWeekKind)
         }
 
-        if HomeWidgetAllTimeData.read() == nil {
+        if HomeWidgetAllTimeData.read() == nil || HomeWidgetAllTimeData.read()?.isEmpty == true {
```

Comment:
@staskus this works but I wonder if it would be cleaner here to create a static wrapper function that returns a Bool?  For example `HomeWidgetTodayData.cacheDataExists()` which when done in HomeWidgetData would only have to be implemented once so that the same / similar logic check is not repeated?

Reply:
Thanks, that's a good idea!